lbry/extras/daemon/analytics.py

```diff
@@ -110,28 +110,33 @@ def __init__(self, conf: Config, installation_id: str, session_id: str):
         self.cookies = {}
         self.url = ANALYTICS_ENDPOINT
         self._write_key = utils.deobfuscate(ANALYTICS_TOKEN)
-        self._enabled = conf.share_usage_data
         self._tracked_data = collections.defaultdict(list)
         self.context = _make_context(system_info.get_platform())
         self.installation_id = installation_id
         self.session_id = session_id
         self.task: typing.Optional[asyncio.Task] = None
         self.external_ip: typing.Optional[str] = None
 
+    @property
+    def enabled(self):
+        return self.conf.share_usage_data
+
     @property
     def is_started(self):
         return self.task is not None
 
     async def start(self):
-        if self._enabled and self.task is None:
-            self.external_ip = await utils.get_external_ip()
+        if self.task is None:
+            self.external_ip = await utils.get_external_ip() if self.enabled else None
             self.task = asyncio.create_task(self.run())
 
     async def run(self):
         while True:
-            await self._send_heartbeat()
+            if self.enabled:
+                await self._send_heartbeat()
             await asyncio.sleep(1800)
-            self.external_ip = await utils.get_external_ip()
+            if self.enabled:
+                self.external_ip = await utils.get_external_ip()
```

Comment:
if the process was started in not enabled state and thus external ip address was not set but then later it was enabled, can `self._send_heartbeat()` work without an external_ip set? it appears above that first heart beat is sent and only after it gets the ip again...